src/components/index.cjsx

```diff
@@ -20,15 +20,26 @@ Dashboard = React.createClass
   displayName: 'Dashboard'
   componentWillMount: -> CurrentUserStore.addChangeListener(@update)
   componentWillUnmount: -> CurrentUserStore.removeChangeListener(@update)
+  
+  contextTypes:
+    router: React.PropTypes.func
 
   update: -> @setState({})
+
   render: ->
     if CurrentUserStore.isCoursesLoaded()
       courses = CurrentUserStore.getCourses()
+      if courses.length is 1
+        roles = courses[0].roles
+        if roles.length is 1
+          singleCourseId = courses[0].id
+          if roles[0].type is 'student'
+            @context.router.transitionTo('viewStudentDashboard', {courseId: singleCourseId})
```

Comment:
That's true. I'm not sure how easy it is to change the handler itself. If it is not that simple then maybe we could replace all the `transitionTo('dashboard', ...)` code with a function that performs this logic of determining if there is only 1 course and then skipping the dashboard and going directly to the student list or teacher calendar.